diff -Nur a/arch/arm/boot/dts/sun8i-h3.dtsi b/arch/arm/boot/dts/sun8i-h3.dtsi
--- a/arch/arm/boot/dts/sun8i-h3.dtsi	2017-10-15 18:33:05.000000000 +0200
+++ b/arch/arm/boot/dts/sun8i-h3.dtsi	2017-10-15 18:39:42.024954230 +0200
@@ -166,6 +166,47 @@
 	};
 
 	soc {
+		sram-controller@01c00000 {
+			compatible = "allwinner,sun4i-a10-sram-controller";
+			reg = <0x01c00000 0x30>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
+			sram_c: sram@01d00000 {
+				compatible = "mmio-sram";
+				reg = <0x01d00000 0xd0000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0 0x01d00000 0xd0000>;
+
+				ve_sram: sram-section@0000 {
+					compatible = "allwinner,sun4i-a10-sram-c1";
+					reg = <0x00000 0x80000>;
+				};
+			};
+		};
+
+		ve: video-engine@01c0e000 {
+			compatible = "allwinner,sun5i-a13-video-engine";
+			syscon = <&syscon>;
+			reg = <0x01c0e000 0x1000>;
+
+			clocks = <&ccu CLK_BUS_VE>,
+			<&ccu CLK_VE>,
+			<&ccu CLK_DRAM_VE>;
+
+			clock-names = "ahb", "mod", "ram";
+
+			assigned-clocks = <&ccu CLK_VE>;
+			assigned-clock-rates = <320000000>;
+
+			resets = <&ccu RST_BUS_VE>;
+
+			interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+			allwinner,sram = <&ve_sram 1>;
+		};
+
 		ths: thermal-sensor@1c25000 {
 			compatible = "allwinner,sun8i-h3-ths";
 			reg = <0x01c25000 0x100>;
