From 6c088282947d21d15e106c9c9ecd69ee32235329 Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime.ripard@bootlin.com>
Date: Wed, 18 Jul 2018 10:10:05 +0200
Subject: [PATCH 55/55] fixup! media: cedrus: Add H264 decoding support

---
 drivers/media/platform/sunxi/cedrus/cedrus_h264.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/platform/sunxi/cedrus/cedrus_h264.c b/drivers/media/platform/sunxi/cedrus/cedrus_h264.c
index 42f700d0c7ea..c7942fe926bf 100644
--- a/drivers/media/platform/sunxi/cedrus/cedrus_h264.c
+++ b/drivers/media/platform/sunxi/cedrus/cedrus_h264.c
@@ -430,6 +430,12 @@ static void cedrus_h264_setup(struct cedrus_ctx *ctx,
 
 	cedrus_engine_enable(dev, CEDRUS_CODEC_H264);
 
+	cedrus_write(dev, VE_H264_SDROT_CTRL, 0);
+	cedrus_write(dev, VE_H264_EXTRA_BUFFER1,
+		     ctx->codec.h264.pic_info_buf_dma - PHYS_OFFSET);
+	cedrus_write(dev, VE_H264_EXTRA_BUFFER2,
+		     (ctx->codec.h264.pic_info_buf_dma - PHYS_OFFSET) + 0x48000);
+
 	cedrus_write_scaling_lists(ctx, run);
 	cedrus_write_frame_list(ctx, run);
 
-- 
2.18.0

